static/downloadables/calculate_score.py

```diff
@@ -4,249 +4,233 @@
     return
 
 
-def txtcalculations(snpSet, txtObj, tableObjDict, isJson, isCondensedFormat, omitUnusedStudiesFile, unmatchedAlleleVariants, clumpedVariants, outputFile, unusedTraitStudy, trait, studyID, snpCount):
+def txtcalculations(snpSet, txtObj, tableObjDict, mafDict, isJson, isCondensedFormat, unmatchedAlleleVariants, clumpedVariants, outputFile, trait, studyID, pValueAnno, betaAnnotation, valueType, snpCount):
+    pValBetaAnnoValType = "|".join((pValueAnno, betaAnnotation, valueType))
+
+    if studyID in tableObjDict['studyIDsToMetaData'].keys():
+        # study info
+        citation = tableObjDict['studyIDsToMetaData'][studyID]['citation']
+        reportedTrait = tableObjDict['studyIDsToMetaData'][studyID]['reportedTrait']
+        betas = [] # holds the oddsRatios used for calculation
+        betaUnits = set() # holds the units for the betas
+        # Output Sets
+        protectiveVariants = set()
+        riskVariants = set()
+
+        mark = False
+
+        nonMissingSnps = 0
+        for snp in txtObj:
+            # Also iterate through each of the alleles
+            if snp in snpSet:
+                nonMissingSnps += 1
+                riskAllele = tableObjDict['associations'][snp]['traits'][trait][studyID][pValBetaAnnoValType]['riskAllele']
+                units = tableObjDict['associations'][snp]["traits"][trait][studyID][pValBetaAnnoValType]['betaUnit']
+                # if the values are betas, then grab the value, if odds ratios, then take the natural log of the odds ratio
+                snpBeta = tableObjDict['associations'][snp]['traits'][trait][studyID][pValBetaAnnoValType]['betaValue'] if valueType == "beta" else math.log(tableObjDict['associations'][snp]['traits'][trait][studyID][pValBetaAnnoValType]['oddsRatio'])
+                for allele in txtObj[snp]:
+                    # Then compare to the gwa study
+                    if allele != "":
+                        if allele == riskAllele:
+                            betas.append(snpBeta)
+                            betaUnits.add(units)
+                            if snpBeta < 0:
+                                protectiveVariants.add(snp)
+                            elif snpBeta > 0:
+                                riskVariants.add(snp)
+                        elif allele == "." :
+                            mafVal = mafDict[snp]['alleles'][riskAllele] if snp in mafDict and riskAllele in mafDict[snp]["alleles"] else 0
+                            betas.append(snpBeta*mafVal)
+                            betaUnits.add(units)
+                            if snpBeta < 0:
+                                protectiveVariants.add(snp)
+                            elif snpBeta > 0:
+                                riskVariants.add(snp)
+                        else:
+                            unmatchedAlleleVariants.add(snp)
+
+        units = betaUnits.pop() if len(betaUnits) == 1 else "NA" #TODO might need to come up with a better way
```

Comment:
Since betaUnits is a set, if the length is more than one, it means that there are multiple units being reported for that unique combination. I was setting it to NA because we shouldn't just use one. This shouldn't be a problem anymore with how we will be filtering. Maybe instead of setting it to NA we should throw an error